### PR TITLE
Avoid hyperref options clash.

### PR DESCRIPTION
**Pull request by adunning**
This is an alternate workaround to <https://github.com/ho-tex/hyperref/issues/25>. The fix in <https://github.com/jgm/pandoc-templates/commit/feffd7c64abab863abd3f6458d1c445d6bfe7fc4> produces an options clash when using a package that also loads `hyperref`, but with different options (such as `pdfx` and `tufte-latex`). This instead loads the `bookmark` package, which provides a more comprehensive solution, as [recommended on StackExchange](https://tex.stackexchange.com/questions/313266/and-non-ascii-characters-in-headings).

While it isn't ideal to add another package to the template, this should be safe to use, since it has been distributed with TeX Live (including BasicTeX) for almost a decade. It would likely make sense to revert to using `hyperref` on its own once the issue has been corrected. The main advantage of the bookmark package is that it does the work in one run instead of two, but I don't believe Pandoc can take advantage of this, since multiple runs are still necessary for cross-references and so forth.

### CODE REVIEW

**Comment by jpcirrus** [comment]
The solution I have been using for this issue in my custom LaTeX templates where hyperref is loaded by other packages, or in this case by the isodoc (letter) document class, is:

```latex
\PassOptionsToPackage{%           need to be set at hyperref package load:
  unicode                       % fix for https://tex.stackexchange.com/questions/313266/and-non-ascii-characters-in-headings
}{hyperref}
\documentclass[$if(lang)$$lang$$else$en-AU$endif$,$if(fontsize)$$fontsize$$else$11pt$endif$,$for(classoption)$$classoption$$sep$,$endfor$]{isodoc}
```

**Comment by wilx** [comment]
It seems to me that we might need something like this to cover document classes that load `hyperref` themselves:

```diff
diff --git a/default.latex b/default.latex
index 3b80d3c..39922c6 100644
--- a/default.latex
+++ b/default.latex
@@ -1,3 +1,9 @@
+\PassOptionsToPackage{hyphens}{url} % url is loaded by hyperref
+$if(colorlinks)$
+\PassOptionsToPackage{usenames,dvipsnames}{color} % color is loaded by hyperref
+$endif$
+\PassOptionsToPackage{unicode=true}{hyperref}
+%
 \documentclass[$if(fontsize)$$fontsize$,$endif$$if(lang)$$babel-lang$,$endif$$if(papersize)$$papersize$paper,$endif$$for(classoption)$$classoption$$sep$,$endfor$]{$documentclass$}
 $if(beamerarticle)$
 \usepackage{beamerarticle} % needs to be loaded first
@@ -60,10 +66,7 @@ $endif$
 $if(geometry)$
 \usepackage[$for(geometry)$$geometry$$sep$,$endfor$]{geometry}
 $endif$
-\usepackage[unicode=true]{hyperref}
-$if(colorlinks)$
-\PassOptionsToPackage{usenames,dvipsnames}{color} % color is loaded by hyperref
-$endif$
+\usepackage{hyperref}
 \hypersetup{
 $if(title-meta)$
             pdftitle={$title-meta$},
```

**Comment by wilx** [comment]
> https://github.com/jgm/pandoc-templates/pull/230#issuecomment-270706394:
> isodoc (letter) document class

I have actually tried using the `default.latex` template with `isodoc` and it did not compile. It choked on `\date{}`.

**Comment by jpcirrus** [comment]
I am using a custom template with the isodoc class.

**Comment by adunning** [comment]
If this can be fixed without another package, so much the better; do you want to submit your solution as a separate pull request, @wilx, or should I modify this one?

**Comment by wilx** [comment]
[**@adunning**](https://github.com/adunning) commented on [28. 1. 2017 14:07 SEČ](https://github.com/jgm/pandoc-templates/pull/230#issuecomment-275847252 "2017-01-28T13:07:51Z - Replied by Github Reply Comments"):
> If this can be fixed without another package, so much the better; do you want to submit your solution as a separate pull request, [@wilx](https://github.com/wilx), or should I modify this one?

Feel free to do this. :)

**Comment by jgm** [comment]
Could someone rebase this onto current master so I can more easily see what the changes are?
A short description of the motivation for the proposed change would also be helpful -- it's a bit hard to follow the thread above.

**Comment by adunning** [comment]
Should be current right now (though the commits should be combined on merge). The motivation is still the same as indicated in the original post: loading `hyperref` with the Unicode option directly produces an options clash if one uses another package that loads it in a different way. This patch avoids this passing such options at the opening.

**Comment by jgm** [comment]
+++ Andrew Dunning [Feb 01 17 04:05 ]:
>   Should be current right now (though the commits should be combined on
>   merge). The motivation is still the same as indicated in the original

Is there a way to combine it prior to merge, so I can
examine it without mentally merging a number of different
commits?

**Comment by wilx** [comment]
[**@jgm**](https://github.com/jgm) commented on [1. 2. 2017 14:25 SEČ](https://github.com/jgm/pandoc-templates/pull/230#issuecomment-276656256 "2017-02-01T13:25:26Z - Replied by Github Reply Comments"):
> <div class="email-fragment">Is there a way to combine it prior to merge, so I can
> examine it without mentally merging a number of different
> commits?</div>

Well, https://github.com/jgm/pandoc-templates/pull/230/files?
